src/lookup/query.ts

```diff
@@ -132,5 +143,20 @@ export async function query(
       )}`,
     );
   }
-  return data;
+
+  if (data.verifiableCredential) {
+    const newVerifiableCredential: (VerifiableCredential & DatasetCore)[] = [];
+    for (let i = 0; i < data.verifiableCredential.length; i += 100) {
+      newVerifiableCredential.push(
+        // eslint-disable-next-line no-await-in-loop
+        ...(await Promise.all(
```

Comment:
```suggestion
      newVerifiableCredential.push(
        // Limit concurrency to avoid memory overflows. For details see
        // https://github.com/inrupt/solid-client-vc-js/pull/849#discussion_r1377400688
        // eslint-disable-next-line no-await-in-loop
        ...(await Promise.all(
```